### PR TITLE
[ENG-5617] Phase 3: Check new `develop` changes

### PR DESCRIPTION
**Pull request by opaduchak**
## Purpose

Sync feature/python-upgrade with newest changes from develop.
Mostly everything is fine. From what We have seen, Uditi’s PR should not bring us conflicts. But John’s PR had several nuances for us and caused several conflicts, as function check_access was replaced by check_resource_permissions, several lines in tests were rewritten using nose (which we do not use now), and some functions were rewritten or split.
For now the conflicts were resolved.

## Changes

- Merged conflict fixes
- fixes to tests which failed due to John's changes

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/browse/ENG-5617

### CODE REVIEW

File: addons/base/views.py [hunk 1]
```diff
@@ -65,6 +65,7 @@
 
 # import so that associated listener is instantiated and gets emails
 from website.notifications.events.files import FileEvent  # noqa
+from osf.utils.requests import requests_retry_session
```
Comment (cslzchen) [inline addons/base/views.py:68]: Can we move this import to after `from osf.utils import permissions` in line 59?

**Review by cslzchen — CHANGES_REQUESTED** [review]
Looks great and 💯 for handling the merge conflicts. A few requests before I am back on Tuesday, thanks!

- [x] Fix trivial import style in `addons/base/views.py`
- [x] Update PR title (ticket number, no longer WIP, etc.)
- [ ]  Add a little bit details to the PR description, this doesn't have to be long, what  @ly-mariia mentioned in Slack to me is good enough
- [ ]  Can any of you verify **locally** that "after this PR is merged in to the feature branch, there is no conflicts merging the feature branch back to the `develop`"?

On question, related to the last item above, I think this https://github.com/CenterForOpenScience/osf.io/commit/35d5512477e02f9d01db2e744c7c41778d42c67b is the actual `merging develop into feature branch` commit that includes conflict fixing. If so, (a note for myself) I will rename it to Merge `develop` into `feature/python-upgrade` and fix conflicts.

Another note for myself: back-up the feature branch before merge, create a dedicated phase 3 branch for deployment after merge.

**Comment by opaduchak** [comment]
> On question, related to the last item above, I think this [35d5512](https://github.com/CenterForOpenScience/osf.io/commit/35d5512477e02f9d01db2e744c7c41778d42c67b) is the actual `merging develop into feature branch` commit that includes conflict fixing. If so, (a note for myself) I will rename it to Merge `develop` into `feature/python-upgrade` and fix conflicts.
Indeed, it is.
And regarding local check, I doubt that this is necessary, because this commit already resolved all conflicts. (I compared develop and this branch on github, and there aren't any conflicts)

**Review by cslzchen — APPROVED** [review]

**Comment by cslzchen** [comment]
Cc @felliott, due to conflicts fix (https://github.com/CenterForOpenScience/osf.io/pull/10618/commits/35d5512477e02f9d01db2e744c7c41778d42c67b) when merging `develop` into the feature branch, `rebase`, `merge --no-ff` and pushing back couldn't work. Now merging this PR in GitHub directly.

**Pull request merged**